hibernation-setup-tool.c

```diff
@@ -1539,15 +1539,13 @@ static int handle_pre_systemd_suspend_notification(const char *action)
                     log_fatal("Couldn't remove /tmp/hibernation-setup-tool directory before proceeding");
                 }
 
-                else {
```

Comment:
The function recursive_rmdir() is suspicious: the return value of nftw() is 0 or -1, which should not be compared with FTW_STOP (1) ?
static bool recursive_rmdir(const char *path) { return nftw(path, recursive_rmdir_cb, 16, FTW_DEPTH | FTW_PHYS | FTW_ACTIONRETVAL) != FTW_STOP; }

Reply:
This took some investigating. `nftw` is a higher order function. `recursive_rmdir_cb` is a callback.

With flag `FTW_ACTIONRETVAL `set, the function `recursive_rmdir_cb` can only return one of: 
```
FTW_CONTINUE (0), 
FTW_STOP (1), 
FTW_SKIP_SUBTREE(2), 
FTW_SKIP_SIBLINGS (3)
```

Based on what the `recursive_rmdir_cb` function returns, `nftw` can either continue walking through the tree or stop.

Since `FTW_ACTIONRETVAL ` is set, after finishing `nftw` can return 0 (on success), -1 (if an error occurred) or in the case we ever threw an `FTW_STOP` in the callback function, FTW_STOP will also get thrown by `nftw`.  

So the idea is, is that we handle all situations an error can occur in the callback and we throw an `FTW_STOP` in all of them. But I think this isn't the safest code, as opposed to checking if we don't throw an `FTW_STOP`, i'm going to change to explicitly check if we get a success. With the current code if `nftw` returns -1 instead of `FTW_STOP`, it would think its a success, when its not. 

`return nftw(path, recursive_rmdir_cb, 16, FTW_DEPTH | FTW_PHYS | FTW_ACTIONRETVAL) == 0;`